include/vcpkg/vcpkgcmdarguments.h

```diff
@@ -29,48 +29,101 @@ namespace vcpkg
         std::vector<std::string> command_arguments;
     };
 
+    struct MetadataMessage
+    {
+        constexpr MetadataMessage() noexcept : kind(MetadataMessageKind::Unused), unused{} { }
+        /*implicit*/ constexpr MetadataMessage(const msg::MessageT<>& message) noexcept
+            : kind(MetadataMessageKind::Message), message(&message)
+        {
+        }
+
+        template<int N>
+        /*implicit*/ constexpr MetadataMessage(const char (&literal)[N]) noexcept
+            : kind(MetadataMessageKind::Literal), literal(literal)
+        {
+        }
+
+        template<class Callback,
+                 std::enable_if_t<std::is_convertible_v<const Callback&, LocalizedString (*)()>, int> = 0>
```

Comment:
For my own education: why does this use `const Callback&` instead of `Callback` by value?

Reply:
I think they are equivalent in this case. I just default to `const T&` unless I have a specific reason to do otherwise when writing templates.

Note that even if the callback is `Callback callback` rather than `const Callback& callback`, the `is_convertible_v` still needs to have a `&` since we are invoking the callback as an lvalue.

Reply:
(Also it's smashed to a function pointer at constexpr time so there's no runtime difference; given that all the other ctors take references I'd like to leave this as-is because overload resolution that involves overloads that bind a reference vs. overloads that do not makes my brain hurt...)